selfservice/strategy/password/validator.go

```diff
@@ -109,9 +120,8 @@ func (s *DefaultPasswordValidator) fetch(hpw []byte, apiDNSName string) error {
 		return errors.Wrapf(ErrUnexpectedStatusCode, "%d", res.StatusCode)
 	}
 
-	s.Lock()
-	s.hashes[b20(hpw)] = 0
-	s.Unlock()
+	s.hashes.SetWithTTL(b20(hpw), 0, 1, hashCacheItemTTL)
+	s.hashes.Wait()
```

Comment:
Is the wait intentional? I think we could do without!

Reply:
Fix in 3c5caa669